sway/commands/seat/hide_cursor.c

```diff
@@ -3,26 +3,48 @@
 #include "sway/commands.h"
 #include "sway/config.h"
 #include "sway/input/seat.h"
+#include "sway/input/cursor.h"
+#include "sway/server.h"
 #include "stringop.h"
+#include "util.h"
 
 struct cmd_results *seat_cmd_hide_cursor(int argc, char **argv) {
 	struct cmd_results *error = NULL;
-	if ((error = checkarg(argc, "hide_cursor", EXPECTED_EQUAL_TO, 1))) {
+	if ((error = checkarg(argc, "hide_cursor", EXPECTED_AT_LEAST, 1))) {
 		return error;
 	}
-	if (!config->handler_context.seat_config) {
+	if ((error = checkarg(argc, "hide_cursor", EXPECTED_AT_MOST, 2))) {
+		return error;
+	}
+	struct seat_config *seat_config = config->handler_context.seat_config;
+	if (!seat_config) {
 		return cmd_results_new(CMD_FAILURE, "No seat defined");
 	}
 
-	char *end;
-	int timeout = strtol(argv[0], &end, 10);
-	if (*end) {
-		return cmd_results_new(CMD_INVALID, "Expected an integer timeout");
-	}
-	if (timeout < 100 && timeout != 0) {
-		timeout = 100;
+	if (argc == 1) {
+		char *end;
+		int timeout = strtol(argv[0], &end, 10);
+		if (*end) {
+			return cmd_results_new(CMD_INVALID, "Expected an integer timeout");
+		}
+		if (timeout < 100 && timeout != 0) {
+			timeout = 100;
+		}
+		seat_config->hide_cursor_timeout = timeout;
+	} else {
+		if (strcmp(argv[0], "keyboard") != 0) {
+			return cmd_results_new(CMD_INVALID,
+				"Expected 'hide_cursor <timeout>|keyboard [enable|disable]'");
```

Comment:
Bikeshed: maybe we could rename the command to `hide_cursor when-typing enable|disable`?

Reply:
Done